decentraland/unity-renderer
# fix: sdk7 tween reset

* The TweenComponent DOTween tweener was being wrongly reset when it wasn't needed, specifically affecting cases of applying different kind of tweens on the same transform one after the other.
* Transform component position was being updated wrongly for entities that were child of other displaced entities instead of the scene root transform.

Solves https://github.com/decentraland/sdk/issues/1063

## TEST INSTRUCTIONS

1. Enter [this playground link](https://playground.decentraland.org/?code=bGV0IHdhaXRpbmdGdW5jdGlvblRpbWVyID0gMApsZXQgd2FpdGluZ0Z1bmN0aW9uCmVuZ2luZS5hZGRTeXN0ZW0oKGR0OiBudW1iZXIpID0%2BIHsKICBpZiAod2FpdGluZ0Z1bmN0aW9uVGltZXIgPD0gMCkgcmV0dXJuCiAgd2FpdGluZ0Z1bmN0aW9uVGltZXIgLT0gZHQKICBpZiAod2FpdGluZ0Z1bmN0aW9uVGltZXIgPD0gMCkgewogICAgd2FpdGluZ0Z1bmN0aW9uKCkKICB9Cn0pCgpmdW5jdGlvbiBjcmVhdGVDdWJlKHg6IG51bWJlciwgeTogbnVtYmVyLCB6OiBudW1iZXIpOiBFbnRpdHkgewogIGNvbnN0IGVudGl0eSA9IGVuZ2luZS5hZGRFbnRpdHkoKQogIFRyYW5zZm9ybS5jcmVhdGUoZW50aXR5LCB7IHBvc2l0aW9uOiB7IHgsIHksIHogfSB9KQogIE1lc2hSZW5kZXJlci5zZXRCb3goZW50aXR5KQogIE1lc2hDb2xsaWRlci5zZXRCb3goZW50aXR5KQogIHJldHVybiBlbnRpdHkKfQoKY29uc3QgaW5pdEVudGl0eSA9IGNyZWF0ZUN1YmUoOCwgMSwgOCkKcG9pbnRlckV2ZW50c1N5c3RlbS5vblBvaW50ZXJEb3duKAogIHsgZW50aXR5OiBpbml0RW50aXR5LCBvcHRzOiB7IGJ1dHRvbjogSW5wdXRBY3Rpb24uSUFfUE9JTlRFUiwgaG92ZXJUZXh0OiAnQ0xJQ0snIH0gfSwKICAoKSA9PiB7CiAgICBtb3ZlVG9SYW5kb21Qb3MoaW5pdEVudGl0eSkKICB9CikKCmV4cG9ydCBmdW5jdGlvbiBtb3ZlVG9SYW5kb21Qb3MoZW50aXR5OiBFbnRpdHkpIHsKICAvLyBEZWZpbmUgc3RhcnQgYW5kIGVuZCBwb3NpdGlvbnMKICBsZXQgc3RhcnRQb3MgPSBUcmFuc2Zvcm0uZ2V0KGVudGl0eSkucG9zaXRpb24KICBsZXQgZW5kUG9zID0gVmVjdG9yMy5jcmVhdGUoTWF0aC5yYW5kb20oKSAqIDEyICsgMSwgMSwgTWF0aC5yYW5kb20oKSAqIDEyICsgMSkKCiAgLy8gRGVmaW5lIHN0YXJ0IGFuZCBlbmQgZGlyZWN0aW9ucwogIGxldCBzdGFydFJvdCA9IFRyYW5zZm9ybS5nZXQoZW50aXR5KS5yb3RhdGlvbgogIGxldCBlbmRSb3QgPSBRdWF0ZXJuaW9uLmZyb21Mb29rQXQoc3RhcnRQb3MsIGVuZFBvcykKCiAgLy8gMS4gUm90YXRlIAogIFR3ZWVuLmNyZWF0ZU9yUmVwbGFjZShlbnRpdHksIHsKICAgIG1vZGU6IFR3ZWVuLk1vZGUuUm90YXRlKHsKICAgICAgc3RhcnQ6IHN0YXJ0Um90LAogICAgICBlbmQ6IGVuZFJvdAogICAgfSksCiAgICBkdXJhdGlvbjogMjAwMCwKICAgIGVhc2luZ0Z1bmN0aW9uOiBFYXNpbmdGdW5jdGlvbi5FRl9FQVNFU0lORQogIH0pCgogIHdhaXRpbmdGdW5jdGlvblRpbWVyID0gMi41CiAgd2FpdGluZ0Z1bmN0aW9uID0gKCkgPT4gewogICAgLy8gMi4gTW92ZQogICAgVHdlZW4uY3JlYXRlT3JSZXBsYWNlKGVudGl0eSwgewogICAgICBtb2RlOiBUd2Vlbi5Nb2RlLk1vdmUoewogICAgICAgIHN0YXJ0OiBzdGFydFBvcywKICAgICAgICBlbmQ6IGVuZFBvcwogICAgICB9KSwKICAgICAgZHVyYXRpb246IDUwMCwKICAgICAgZWFzaW5nRnVuY3Rpb246IEVhc2luZ0Z1bmN0aW9uLkVGX0VBU0VTSU5FCiAgICB9KQoKICAgIHdhaXRpbmdGdW5jdGlvbiA9ICgpID0%2BIHsKICAgICAgLy8gMy4gU2NhbGUKICAgICAgVHdlZW4uY3JlYXRlT3JSZXBsYWNlKGVudGl0eSwgewogICAgICAgIG1vZGU6IFR3ZWVuLk1vZGUuU2NhbGUoewogICAgICAgICAgc3RhcnQ6IFRyYW5zZm9ybS5nZXQoZW50aXR5KS5zY2FsZSwKICAgICAgICAgIGVuZDogVmVjdG9yMy5zY2FsZShUcmFuc2Zvcm0uZ2V0KGVudGl0eSkuc2NhbGUsIDEgKyAoTWF0aC5yYW5kb20oKSAqIDAuNSAqIChNYXRoLnJhbmRvbSgpID4gMC41PyAxIDogLTEpKSkKICAgICAgICB9KSwKICAgICAgICBkdXJhdGlvbjogNTAwLAogICAgICAgIGVhc2luZ0Z1bmN0aW9uOiBFYXNpbmdGdW5jdGlvbi5FRl9FQVNFU0lORQogICAgICB9KQogICAgfQogICAgd2FpdGluZ0Z1bmN0aW9uVGltZXIgPSAxCiAgfQp9&explorer-branch=fix%2Fsdk7-tween-reset) and wait for the world to load
2. Hit the RUN button
3. Confirm that the clickable cube executes this expected **sequential behaviour:** ROTATE + MOVE + SCALE and that the position of the cube doesn't get reset/flickered when the scaling starts or when clicking the cube again after the sequence. (Since the scaling is also randomized, if you don't see it scaling it may be because the randomized values were close to the current ones, just use the cube again)
4. After that sequence, click the cube again and confirm it keeps working
